contracts/SP1_SpokePool.sol

```diff
@@ -0,0 +1,167 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.0;
+
+import "./SpokePool.sol";
+
+interface IHelios {
+    function executionStateRoots(uint256) external view returns (bytes32);
+}
+
+interface ISP1Verifier {
+    function verifyProof(
+        bytes32 programVKey,
+        bytes calldata publicValues,
+        bytes calldata proofBytes
+    ) external view;
+}
+
+/**
+ * @notice SP1 Spoke pool capable of receiving data stored in L1 state via SP1 + Helios light clients.
+ */
+contract SP1_SpokePool is SpokePool {
+    // The public values stored on L1 that can be relayed into this contract when accompanied with an SP1 proof.
+    struct ContractPublicValues {
+        bytes32 stateRoot;
```

Comment:
I think the `stateRoot` existing in ContractPublicValues prevents a lot of replay-attack vectors but still allows for the case where the `message` destined for this SpokePool is legimately the same as some previous message. For example, the `pauseFills()` calldata would be the same multiple times if wanted to pause-unpause-pause-unpause for some reason over the course of X months.